modules/apps/headless/headless-admin-workflow/headless-admin-workflow-test/src/testIntegration/java/com/liferay/headless/admin/workflow/resource/v1_0/test/WorkflowLogResourceTest.java

```diff
@@ -15,14 +15,388 @@
+import com.liferay.portal.kernel.test.rule.DataGuard;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.workflow.WorkflowTaskManager;
+import com.liferay.portal.test.rule.Inject;
 
-import org.junit.Ignore;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
 import org.junit.runner.RunWith;
 
 /**
  * @author Javier Gamarra
  */
-@Ignore
+@DataGuard(scope = DataGuard.Scope.METHOD)
 @RunWith(Arquillian.class)
 public class WorkflowLogResourceTest extends BaseWorkflowLogResourceTestCase {
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		BaseWorkflowLogResourceTestCase.setUpClass();
+
+		_resourceBundleLoader =
+			ResourceBundleLoaderUtil.getPortalResourceBundleLoader();
+
+		_workflowDefinition =
+			WorkflowDefinitionTestUtil.addWorkflowDefinition();
+
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(
+				UserLocalServiceUtil.getUser(TestPropsValues.getUserId())));
+	}
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_siteContentReviewerRole = _roleLocalService.getRole(
+			testGroup.getCompanyId(), RoleConstants.SITE_CONTENT_REVIEWER);
+
+		_workflowInstance = WorkflowInstanceTestUtil.addWorkflowInstance(
+			testGroup.getGroupId(), ObjectReviewedTestUtil.addObjectReviewed(),
+			_workflowDefinition);
+	}
+
+	@Override
+	@Test
+	public void testGetWorkflowInstanceWorkflowLogsPage() throws Exception {
+		WorkflowTask workflowTask = WorkflowTaskTestUtil.getWorkflowTask(
+			_workflowInstance.getId());
+
+		_workflowTaskManager.assignWorkflowTaskToUser(
+			TestPropsValues.getCompanyId(), TestPropsValues.getUserId(),
+			workflowTask.getId(), TestPropsValues.getUserId(), StringPool.BLANK,
+			null, null);
+
+		Page<WorkflowLog> page =
+			workflowLogResource.getWorkflowInstanceWorkflowLogsPage(
+				_workflowInstance.getId(),
+				new String[] {WorkflowLog.Type.TRANSITION.getValue()},
+				Pagination.of(1, 2));
+
+		Assert.assertEquals(0, page.getTotalCount());
+
+		page = workflowLogResource.getWorkflowInstanceWorkflowLogsPage(
+			_workflowInstance.getId(),
+			new String[] {WorkflowLog.Type.TASK_ASSIGN.getValue()},
+			Pagination.of(1, 2));
+
+		Assert.assertEquals(2, page.getTotalCount());
+
+		assertEquals(
+			Arrays.asList(
+				new WorkflowLog() {
+					{
+						commentLog = StringPool.BLANK;
+						description = _language.format(
+							_getResourceBundle(),
+							"x-assigned-the-task-to-himself",
+							new Object[] {
+								_portal.getUserName(
+									TestPropsValues.getUserId(),
+									StringPool.BLANK),
+								_portal.getUserName(
+									TestPropsValues.getUserId(),
+									StringPool.BLANK)
```

Comment:
Hi, @seltong

for this property `x-assigned-the-task-to-himself` we need only one parameter.